### PR TITLE
feat(indent-blankline): update to v3, add scope color

### LINKED ISSUE #583: Support indent-blankline 3.0
indent-blankline released a new version with a lot of changes so the catppuccin integration seems to not work anymore.

See: https://github.com/lukas-reineke/indent-blankline.nvim/wiki/Migrate-to-version-3 for migration guide.

### PR DESCRIPTION
**Pull request by Pandoks**
Updated [indent-blankline](https://github.com/lukas-reineke/indent-blankline.nvim) to support their latest version.

Added feature to let users choose what color they want for their scope indent line.

Fixes #583

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -680,6 +692,7 @@ hop = false
 ```lua
 indent_blankline = {
     enabled = true,
+	scope_color = "", -- catppuccin color (eg. lavendar) Default: text
```
Comment (nullchilly) [inline README.md:695]: Wrong indentation, what did you use to format readme.md?

Comment (Pandoks) [reply README.md]: Prettier. I turned my formatter off the the most recent commit.

File: lua/catppuccin/types.lua [hunk 2]
```diff
@@ -222,6 +222,8 @@
 ---@class CtpIntegrationIndentBlankline
 -- Whether to enable the integration.
 ---@field enabled boolean
+-- Sets the color of the scope line
+---@field scope_color string?
```
Comment (nekowinston) [inline lua/catppuccin/types.lua:226]: Oh sorry I missed that this [*has to be a Catppuccin color*](https://github.com/catppuccin/nvim/blob/2fe1aa6a19ae6b5346da3ee0211d87685b6faf7d/lua/catppuccin/groups/integrations/indent_blankline.lua#L10) before.
Let's please use
```suggestion
---@field scope_color CtpColor?
```
Should be good to merge after that.

**Review by nekowinston — APPROVED** [review]

**Comment by Pandoks** [comment]
> The link in the README should now link to https://github.com/lukas-reineke/indent-blankline.nvim#multiple-indent-colors for an explain on how to set up `colored_indent_levels`. The current link is outdated.

_Originally posted by @schemar in https://github.com/catppuccin/nvim/issues/584#issuecomment-1740583810_

**Comment by nekowinston** [comment]
Thanks for the PR! :)

**Pull request merged**